test/extensions/filters/common/ext_authz/ext_authz_grpc_impl_test.cc

```diff
@@ -436,6 +437,46 @@ TEST_F(ExtAuthzGrpcClientTest, AuthorizationOkWithAppendActions) {
   client_->onSuccess(std::make_unique<envoy::service::auth::v3::CheckResponse>(check_response),
                      span_);
 }
+// Test the client when an error code response is received.
+TEST_F(ExtAuthzGrpcClientTest, AuthorizationReturnsErrorOnBadGrpcCode) {
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues(
+      {{"envoy.reloadable_features.process_ext_authz_grpc_error_codes_as_errors", "true"}});
```

Comment:
Please add a second test case that verifies this fails when the runtime guard is set to false.